File: TD/TD3/Code/5.cpp
/* Calculer la somme des N premiers entiers */
#include <iostream>

int calcSomme(int a)
{
  int b;

  b = 0;

  for (int i = 0; i <= a; ++i) {
    b += i;
  }
  
  // On peut utiliser une autre méthode plus facile //
  // sans boucle POUR , c'est d'utiliser la formule //
  //            b = a * (a + 1) / 2                 //
  // Dans ce cas là il faut juste initialiser b ,   //
  // sans la donnant une valeur !                   //
  
  return b;
}

int main()
{
  int a;
  
  std::cout << "Donne moi une valeur : " << std::flush;
  std::cin >> a;
  
  std::cout << "La somme de " << a << " premiers entiers est : " << calcSomme(a);
  
  std::cout << std::endl;
  return 0;
}

File: TD/TD3/Code/Python/5.py
#!/usr/bin/python
# Calculer la somme des N premiers entiers


def calc_somme(n):
    somme = 0

    for i in range(n + 1):
        somme += i

    # On peut utiliser une autre méthode plus facile #
    # sans boucle POUR , c'est d'utiliser la formule #
    #            somme = n * (n + 1) / 2             #
    #                      FIN!                      #

    return somme


print('Donne moi une valeur : ', end='')
val = int(input())

print('La somme de', val, 'premiers entiers',
      'est', calc_somme(val))
